extensions/librdkafka/PublishKafka.cpp

```diff
@@ -99,6 +99,13 @@ core::Property PublishKafka::MessageKeyField("Message Key Field", "The name of a
                                              "");
 core::Property PublishKafka::DebugContexts("Debug contexts", "A comma-separated list of debug contexts to enable."
                                            "Including: generic, broker, topic, metadata, feature, queue, msg, protocol, cgrp, security, fetch, interceptor, plugin, consumer, admin, eos, all", "");
+const core::Property PublishKafka::FailEmptyFlowFiles(
+    core::PropertyBuilder::createProperty("Fail empty flow files")
+        ->withDescription("Keep backwards compatibility with <=0.7.0 bug which caused flow files with empty content to not be published to Kafka and forwarded to failure. The old behavior is "
+                          "deprecated. Use connections to drop empty flow files!")
+        ->isRequired(false)
```

Comment:
I think it should be required.

Reply:
If we make it required, old flows missing the property will fail, so we will not be backwards compatible.

Reply:
Fair point.
Having a default value while the property is not required is always a bad smell for me. In this case it's justified.

Reply:
Does a default value satisfy the requirement for the property? If so, then I can set `isRequired(true)` and it would still keep old flows working

Reply:
In our perspective, it does, as whenever we get the value, there is a valid value. 

Required would force the user to specify something, in this case it doesn't apply, so it ensures backward compatibility as you highlighted.